tests/DataTest.php

```diff
@@ -2278,3 +2279,31 @@ public function __construct(
     expect($invaded->_except)->toBeEmpty();
     expect($invaded->_wrap)->toBeNull();
 });
+
+it('can fetch lazy union data', function () {
+    $data = UnionData::from(1);
+
+    expect($data->id)
```

Comment:
I would make separate expectations. Imho the `and` makes it less readable to see what is actually being tested.